temporalio/bridge/client.py

```diff
@@ -0,0 +1,67 @@
+import os
+import socket
+from dataclasses import dataclass, field
+from typing import Mapping, Optional, Type, TypeVar
+
+import google.protobuf.message
+import temporal_sdk_bridge
+
+import temporalio.api.workflowservice.v1
+
+
+@dataclass
+class ClientTlsConfig:
+    server_root_ca_cert: Optional[bytes]
+    domain: Optional[str]
+    client_cert: Optional[bytes]
+    client_private_key: Optional[bytes]
+
+
+@dataclass
+class ClientRetryConfig:
+    initial_interval_millis: int = 100
+    randomization_factor: float = 0.2
+    multiplier: float = 1.5
+    max_interval_millis: int = 5000
+    max_elapsed_time_millis: Optional[int] = 10000
+    max_retries: int = 10
+
+
+@dataclass
+class ClientOptions:
+    target_url: str
+    client_name: str = "temporal-python"
+    # TODO(cretz): Take from importlib ref https://stackoverflow.com/a/54869712
+    client_version: str = "0.1.0"
+    static_headers: Mapping[str, str] = field(default_factory=dict)
+    identity: str = f"{os.getpid()}@{socket.gethostname()}"
+    # TODO(cretz): Use proper name/version
+    worker_binary_id: str = "python-sdk@0.1.0"
```

Comment:
This todo should change to have it end up being specific to what user code is included (combined with sdk version). `client_name` and `client_version` are sufficient for what this currently has in it.

Reply:
I took this from https://github.com/temporalio/sdk-typescript/blob/67526d3a9effaea51d9aac3573ba2f8c9d3e1bab/packages/worker/src/server-options.ts#L53. Does that need to change to also be user-code-specific?

In Go we do an md5 of the running executable (which isn't that right either for those using `go run`), but I'm unsure how to get a unique hash of running Python code.

Reply:
Yeah, it absolutely should. Didn't realize TS wasn't doing this the right way either.

Not clear to me what the technique should be either but I'm guessing there's something we can hack together.

Reply:
I copied it from Java and later opened https://github.com/temporalio/sdk-typescript/issues/204

Reply:
I am still investigating how best to hash the unique code set in Python without being too burdensome/flaky. I will watch the TS issue to see if they do anything beyond hashing the bundle they already have.

Reply:
After further discussion, we're just gonna crawl and hash the imports if the caller hasn't explicitly provided a value